stl/inc/format

```diff
@@ -2541,8 +2541,8 @@ _NODISCARD _OutputIt _Fmt_write(
         case chars_format::hex:
         case chars_format::scientific:
             if (_Extra_precision != 0) {
-                while (*_Exponent_start != _Exponent) { // Trailing zeroes are in front of the exponent
-                    --_Exponent_start;
+                // Trailing zeroes are in front of the exponent
+                while (*--_Exponent_start != _Exponent) {

```

Comment:
Should we `_STL_INTERNAL_CHECK(_Exponent_start != _Buffer);` in this `while` loop, or is that too paranoid?

Reply:
I think that's unnecessarily paranoid. There's a `_STL_ASSERT(_Result.ec == errc{}, "to_chars failed");` (which is active for users, even though it could be internal) making sure that `to_chars` has succeeded. Given that, and the other branches I mentioned (value finite, hex/scientific), we know the loop will succeed. (Finite is important, since `"infinity"` and `"nan"` wouldn't contain an exponent character.)

Reply:
I'm sold.